src/wmtk/operations/OperationFactory.hpp

```diff
@@ -16,17 +16,31 @@ class OperationFactoryBase
     PrimitiveType m_primitive;
 };
 
-template <typename OperationType>
+template <typename OperationType, typename SettingsType = std::nullptr_t>
 class OperationFactory : public OperationFactoryBase
 {
 public:
     OperationFactory()
         : OperationFactoryBase(OperationType::primitive_type())
+        , m_settings(nullptr)
     {}
+
+    OperationFactory(const SettingsType& settings)
+        : OperationFactoryBase(OperationType::primitive_type())
+        , m_settings(settings)
+    {}
+
     std::unique_ptr<Operation> create(Mesh& m, const Tuple& t) const override
     {
-        return std::make_unique<OperationType>(m, t);
+        if constexpr (std::is_same<SettingsType, std::nullptr_t>::value) {
```

Comment:
I changed it in [this commit](https://github.com/wildmeshing/wildmeshing-toolkit/pull/344/commits/57e65265a7619c71a909cec2e1a3796004fdbe6f). Let me know if you like that better. I think it's already much cleaner but I assume that it can be done in a nicer way.

Reply:
looks a lot nicer! good stuff